### PR TITLE
RFC for Datasource Attribute

### PR DESCRIPTION
**Pull request by jayaranigarg**

### CODE REVIEW

File: RFCs/007-DataSource-Attribute.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+## Detailed Design
+
+### Requirements
+1. DataSource and ITestDataSource should have a common code flow.
+2. DataSource should provide the data for that invocation in the TestContext object.
+3. Design should be extensible to support in-assembly parallelization on a data source.
+
+### Proposed solution
+The test adapter should define an interface class `ITestDataSource` (on similar lines of framework's ITestdatSource interface)which can be extended to get data from custom data source.
```
Comment (harshjain2) [inline RFCs/007-DataSource-Attribute.md]: nit : ITestdatSource  -> ITestDataSource

File: RFCs/007-DataSource-Attribute.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        /// <param name="testMethodInfo">
+        /// The info of test method.
+        /// </param>
+        /// <param name="testContext">
+        /// Test Context object
+        /// </param>
+        /// <returns>
+        /// Test data for calling test method.
+        /// </returns>
+        IEnumerable<object> GetData(UTF.ITestMethod testMethodInfo, ITestContext testContext);
```
Comment (harshjain2) [inline RFCs/007-DataSource-Attribute.md]: Are we removing `GetDisplayName`

```csharp
string GetDisplayName(MethodInfo methodInfo, object[] data);
```

File: RFCs/007-DataSource-Attribute.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+2. DataSource should provide the data for that invocation in the TestContext object.
+3. Design should be extensible to support in-assembly parallelization on a data source.
+
+### Proposed solution
+The test adapter should define an interface class `ITestDataSource` (on similar lines of framework's ITestdatSource interface)which can be extended to get data from custom data source.
+```csharp
+    /// <summary>
+    /// Interface that provides values from data source when data driven tests are run.
+    /// </summary>
+    public interface ITestDataSource
```
Comment (AbhitejJohn) [inline RFCs/007-DataSource-Attribute.md]: add a namespace to say that this is an adapter interface,

**Comment by AbhitejJohn** [comment]
DataSource Vs ITestDataSource
- Differences in their behavior and **why** they are conceptually different workflows.
- Changes in behavior in the Data Driven workflows in general - Calling Execute for each DataRow as opposed to calling it once for all DataRows.
- Support scenarios - DataSource does not work with DataRow, DataSource cannot be extended.

**Pull request merged**